vision/segmentation/plane_segmentation.go

```diff
@@ -191,24 +203,27 @@ type PlaneSegmentation interface {
 }
 
 type pointCloudPlaneSegmentation struct {
-	cloud       pc.PointCloud
-	threshold   float64
-	minPoints   int
-	nIterations int
+	cloud             pc.PointCloud
+	distanceThreshold float64
+	minPoints         int
+	nIterations       int
+	angleThreshold    float64
 }
 
 // NewPointCloudPlaneSegmentation initializes the plane segmentation with the necessary parameters to find the planes
 // threshold is the float64 value for the maximum allowed distance to the found plane for a point to belong to it.
 // minPoints is the minimum number of points necessary to be considered a plane.
-func NewPointCloudPlaneSegmentation(cloud pc.PointCloud, threshold float64, minPoints int) PlaneSegmentation {
-	return &pointCloudPlaneSegmentation{cloud, threshold, minPoints, 2000}
+func NewPointCloudPlaneSegmentation(cloud pc.PointCloud, distanceThreshold float64, minPoints int,
```

Comment:
I think this function may need to be refactored - finding the planes, and finding the ground plane, are two distinct functions of the the algorithm. Sometimes you'll want to find all planes, and sometimes you'll just want to find the ground plane.